include/ICPUBuffer.h

```diff
@@ -37,10 +37,18 @@ class ICPUBuffer : public IBuffer
             size = sizeInBytes;
         }
 
+		//! Returns buffer's type which in this case is always equal to `E_BUFFER_TYPE::EBT_UNSPECIFIED_BUFFER`.
         virtual E_BUFFER_TYPE getBufferType() const {return EBT_UNSPECIFIED_BUFFER;}
-        //! size in BYTES
+
+        //! Returns size in bytes.
         virtual const uint64_t& getSize() const {return size;}
 
+		//! Reallocates internal data. Invalidate any sizes, pointers etc. returned before!
+		/** @param newSize New size of memory.
+		@param forceRetentionOfData Doesn't matter.
```

Comment:
Doesn't matter right now, but if we switch to a memory allocator that does not offer a free `realloc` then this will necessitate a memcpy.

Reply:
It's one of things I wasn't sure about meaning. I'll ask you about this later anyway :)